docs/experiment.md

```diff
@@ -0,0 +1,8 @@
+# Experiment
+
+## Overview
```

Comment:
Feels a bit weird to have empty sections like this & `Common Conventions` below, and I think there are things we could write in these sections, but i think it's fine to leave them blank for now and just focus on the most concrete documentation while we still work in implementation